### ISSUE TITLE
datavariable table name column varchar(255) cause Exceptions for longer tabular datas

### ISSUE THREAD

**Issue by krminta** [IQSS/dataverse#7386]
@tparkola @4tikhonov
Dataverse v. 4.19 build 331-affbf4f.
I wanted to upload some xlsx file, but i got an error:

> "Tabular data ingest failed. Ingest produced tabular data, but failed to save it in the database; Transaction aborted No further information is available."

After checking out the log file, I found out that one of my tabs got a cell, which was treated as header, and max length for header is 255 (column type varchar(255)) . Mine was longer.
I got through this by changing _datavariable_  name column type to TEXT. 
`ALTER TABLE datavariable ALTER COLUMN name TYPE TEXT;`
Then it worked just fine.

**Comment by pdurbin** [comment]
@krminta hi, if it's safe for your xlsx file to be public can you please link to it? I believe there are some other open issues about ingesting Excel documents and maybe your issue could be investigated at the same time (whenever that is).

**Comment by krminta** [comment]
Hi!
I`m not sure if I can share original file, so I made up a file that causes the same error.
[dataverse_faulty_header.xlsx](https://github.com/IQSS/dataverse/files/5493596/dataverse_faulty_header.xlsx)

**Comment by pdurbin** [comment]
@krminta thanks, I understand better what's going on now. I'll put a screenshots below to remind me.

Would a workaround be to add a header to the file?

<img width="287" alt="Screen Shot 2020-11-06 at 10 53 43 AM" src="https://user-images.githubusercontent.com/21006/98386635-6de89980-201e-11eb-80a7-8c22ac96389b.png">

**Comment by krminta** [comment]
Yeah, exactly. It's just that, when there is no header, data is interpreted as a header. So adding a shorter header will do the trick.

**Comment by pdurbin** [comment]
@krminta gotcha. I'm not going to try hard to convince you to add a header but I will point out that variable names are searchable in Dataverse. Variable labels (descriptions) are also searchable if you use a rich format like Stata or Rdata or fill them in afterward using the Data Curation Tool. So you might want to consider adding a header row.

If you don't want to add a header row and would prefer having data be ingested without a header row, can you change the title of this issue to reflect this?

**Comment by krminta** [comment]
@pdurbin done!

Actually adding a header is the best solution, it was just annoying that at first I couldn't pass the data processing of said file. And if not for my access for logs, I think I wouldn't be able to pass it for  much longer time.

**Comment by pdurbin** [comment]
@krminta um, a couple years later, I'm not sure I fully understand your last reply. Adding a header is the best solution, you say. Great! Does that mean you're ok with closing this issue?

**Comment by bencomp** [comment]
This reminds me of #2626 :)

**Comment by bencomp** [comment]
This reminds me of #2626 :)
